src/main/kotlin/de/smartsquare/starter/mqtt/MqttPublisher.kt

```diff
@@ -37,17 +45,39 @@ class Mqtt5Publisher(private val adapter: MqttMessageAdapter, client: Mqtt5Clien
 
     private val asyncClient = client.toAsync()
 
+    data class PublishingOptions(
+        val retain: Boolean = false,
+        val messageExpiryInterval: Long = MqttPublish.NO_MESSAGE_EXPIRY,
+        val payloadFormatIndicator: Mqtt5PayloadFormatIndicator? = null,
+        val contentType: MqttUtf8StringImpl? = null,
+        val responseTopic: MqttTopicImpl? = null,
+        val correlationData: ByteBuffer? = null,
+        val userProperties: MqttUserPropertiesImpl = MqttUserPropertiesImpl.NO_USER_PROPERTIES,
```

Comment:
I think these types and defaults would be better to have a more ergonomic API and to avoid exposing types from the `internal` package of the client.

```suggestion
        val contentType: String? = null,
        val responseTopic: String? = null,
        val correlationData: ByteArray? = null,
        val userProperties: Mqtt5UserProperties  = Mqtt5UserProperties.of(),
```

Reply:
Seems like I already was at sleep when writing this lol. Obviously not exposing internal types. 
I did change one thing though, in favour of using `val responseTopic: String?` I used `val responseTopic: MqttTopic?`